### DISCUSSION TITLE
Unable to start webserver with breeze fresh install

### DISCUSSION THREAD

**Discussion by goncalorazevedo** [apache/airflow#39792] [Q&A] [answered]
I am getting this error when running `breeze start-airflow` after installing it following the steps provided [here](https://github.com/apache/airflow/blob/main/dev/breeze/doc/01_installation.rst)
```
[4/4] Building fresh packages...
error /home/user/new/airflow/airflow/www/node_modules/core-js-pure: Command failed.
Exit code: 126
Command: node -e "try{require('./postinstall')}catch(e){}"
Arguments: 
Directory: /home/user/new/airflow/airflow/www/node_modules/core-js-pure
Output:
/bin/sh: /tmp/yarn--1716484278052-0.6590744740398318/node: /bin/sh: bad interpreter: Text file busy
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Traceback (most recent call last):
  File "/home/user/new/airflow/./scripts/ci/pre_commit/compile_www_assets.py", line 71, in <module>
    subprocess.check_call(["yarn", "install", "--frozen-lockfile"], cwd=os.fspath(www_directory))
  File "/usr/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['yarn', 'install', '--frozen-lockfile']' returned non-zero exit status 126.
```

This is the state of docker before running breeze
```
> docker images --all
REPOSITORY    TAG       IMAGE ID       CREATED         SIZE
hello-world   latest    d2c94e258dcb   12 months ago   13.3kB
```
```
> docker ps --all
CONTAINER ID   IMAGE         COMMAND    CREATED       STATUS                   PORTS     NAMES
553be867b2f9   hello-world   "/hello"   8 weeks ago   Exited (0) 8 weeks ago             charming_hawking
```
Thank you in advance!

Edit: If I pass the `--skip-assets-compilation` I can get the webserver running, however it doesn't seem to be working correctly, clicking the Cluster Activity tab just opens a blank page and a lot of 404 are happening on every page.

**Comment by goncalorazevedo** [comment]
To update on this issue, I have tried on a new system, using EndeavourOS (Arch Linux), kernel 6.9.1-arch1-2 and the issues persists, I also tried running `breeze compile-www-assets`  beforehand.

**Reply by goncalorazevedo** [reply]
I believe this might be an issue with my nodejs and/or yarn version, currently I don't know how to fix it. However, if anyone faces a similar problem, I got around it by configuring my airflow editable installation (ended up not using breeze) to the same version of the PyPI airflow package, currently v2.9 (installed my local package from the v2-9-stable branch) and simply copied the `static` folder from the PyPI distribution into my airflow source.

**Reply by Taragolis** [reply]
I guess it is related to https://github.com/yarnpkg/berry/issues/6309

**Reply by Taragolis** [reply, marked as answer]
Try to install node via ~[`nodeenv`](https://github.com/ekalinin/nodeenv)~ [`nodenv`](https://github.com/nodenv/nodenv) or [`nvm`](https://github.com/nvm-sh/nvm)

**Reply by goncalorazevedo** [reply]
> Try to install node via ~[`nodeenv`](https://github.com/ekalinin/nodeenv)~ [`nodenv`](https://github.com/nodenv/nodenv) or [`nvm`](https://github.com/nvm-sh/nvm)

Hi! Thank you very much, this was definitely the problem! Installing nodejs 22.2.0 via `nvm` now let's me build the www assets. However, I had to do it manually, running `breeze start-airflow` still results in the same issue even with the correct node version activated, I don't know what may be causing it.

Thank you again!

**Reply by Taragolis** [reply]
The only differences that i could see `breeze start-airflow` that breeze start compiles in background and `breeze compile-www-assets` not.

**Reply by potiuk** [reply]
PR fixing it here:  https://github.com/apache/airflow/pull/39824 `breeze compile-www-assets` uses pre-commit to setup and run node and version of node used is set in .pre-commit-config.yml
